jkq/ddsim/pathqasmsimulator.py

```diff
@@ -0,0 +1,271 @@
+        }
+        super().__init__(configuration=configuration or BackendConfiguration.from_dict(conf), provider=provider)
+
+    def run(self, quantum_circuits: Union[QuantumCircuit, List[QuantumCircuit]], **options):
+        if isinstance(quantum_circuits, QasmQobj) or isinstance(quantum_circuits, PulseQobj):
+            raise QiskitError('QasmQobj and PulseQobj are not supported.')
+
+        if not isinstance(quantum_circuits, list):
+            quantum_circuits = [quantum_circuits]
+
+        out_options = {}
+        for key in options:
+            if not hasattr(self.options, key):
+                warnings.warn("Option %s is not used by this backend" % key, UserWarning, stacklevel=2)
+            else:
+                out_options[key] = options[key]
+        circuit_qobj = assemble(quantum_circuits, self, **out_options)
+
+        job_id = str(uuid.uuid4())
+        local_job = JKQJob(self, job_id, self._run_job, circuit_qobj, **options)
+        local_job.submit()
+        return local_job
+
+    def _run_job(self, job_id, qobj_instance: Qobj, **options):
+        self._validate(qobj_instance)
+
+        start = time.time()
+        result_list = [self.run_experiment(qobj_exp, **options) for qobj_exp in qobj_instance.experiments]
+        end = time.time()
+
+        result = {'backend_name': self.configuration().backend_name,
+                  'backend_version': self.configuration().backend_version,
+                  'qobj_id': qobj_instance.qobj_id,
+                  'job_id': job_id,
+                  'results': result_list,
+                  'status': 'COMPLETED',
+                  'success': True,
+                  'time_taken': (end - start),
+                  'header': qobj_instance.header.to_dict()
+                  }
+        return Result.from_dict(result)
+
+    def run_experiment(self, qobj_experiment: QasmQobjExperiment, **options):
+        start_time = time.time()
+        seed = options.get('seed', -1)
+        mode = options.get('mode', 'sequential')
+        nthreads = int(options.get('nthreads', 1))
+        # TODO: Add new strategies here
+        if mode == 'sequential':
+            task_based_mode = ddsim.PathSimulatorMode.sequential
+        elif mode == 'pairwise_recursive':
+            task_based_mode = ddsim.PathSimulatorMode.pairwise_recursive
+        elif mode == 'cotengra':
+            task_based_mode = ddsim.PathSimulatorMode.cotengra
+        elif mode == 'bracket':
+            task_based_mode = ddsim.PathSimulatorMode.bracket
+        elif mode == 'alternating':
+            task_based_mode = ddsim.PathSimulatorMode.alternating
+        else:
+            raise JKQSimulatorError('Simulation mode', mode,
+                                    'not supported by JKQ path simulator. Available modes are \'sequential\', \'pairwise_recursive\', \'cotengra\', \'bracket\' and \'alternating\'')
+
+        sim = ddsim.PathCircuitSimulator(qobj_experiment, seed, task_based_mode, nthreads)
```

Comment:
Once configuration of parameters is implemented properly on the C++ side, it should also be made available here.